eIDClientCore/bin/SimpleClient/Makefile.am

```diff
@@ -1,8 +1,12 @@
 bin_PROGRAMS = SimpleClient
+lib_LTLIBRARIES = libSimpleClient.la
+
 
 noinst_HEADERS = CeIdObject.h CivetServer.h civetweb.h eidui_cli.h
 
 SimpleClient_SOURCES  = main.cpp eidui_cli.cpp CeIdObject.cpp CivetServer.cpp civetweb.c
+libSimpleClient_la_SOURCES = $(SimpleClient_SOURCES)
```

Comment:
why don't you try compiling the other way around?
1. compile your qt gui as library (e.g. `libeiduiqt.so`) that exports the callback functions
2. compile the existing command line UI as library (e.g. `libeiduicli.so`) that exports the callback functions
3. compile simple client as program that either links against `libeiduiqt.so` or `libeiduicli.so`